File: tools/get_dg_matches.pl
#!/usr/bin/perl -w
use strict;

# Jon Nall
# nall AT themountaingoats DOT net
# 06.01.04
#
# Released under the GNU GPLv2 License


$main::GAME_DIR = "$ENV{HOME}/dg_matches";
$main::USER="";          # USERID from your cookies file
$main::PASS=""; # PASSWORD from your cookies file

$main::DG_BASE = "http://www.dailygammon.com/bg/user/";
$main::DG_REVIEW_BASE = "http://www.dailygammon.com/bg/export/";
$main::WGET = "/usr/bin/wget --header='Cookie: USERID=$main::USER; PASSWORD=$main::PASS'";


sub main()
{
    if ($main::USER eq "" || $main::PASS eq "")
    {
        printf(STDERR "Please set USER and PASS at the top of this script.\n");
        printf(STDERR "These values are most likely in your browser's cookies file\n");
        exit(1);
    }
    my $gameref = &getMatchNumbers();
    my @games = @{$gameref};

    mkdir($main::GAME_DIR);
    foreach my $id (@games)
    {
        printf("Retrieving Game# $id\n");
        system("$main::WGET -O '${main::GAME_DIR}/${id}.mat' '${main::DG_REVIEW_BASE}/$id' > /dev/null 2>&1");
    }
}


sub getMatchNumbers()
{
    my @games = ();

    my @results = `$main::WGET -O '-' '${main::DG_BASE}/${main::USER}?finished=1' 2> /dev/null`;

    foreach my $line (@results)
    {
       if($line =~ /<A href=(.*?)>Review<\/A>/) {
           my $game_url = $1;
           $game_url =~ /\/bg\/game\/([0-9]+)\//;
           push(@games, $1);
        }
    } 

    return \@games;
}

main::main();

File: tools/get_dg_matches.php
<?php
/**
 * Get's all the matches from dailygammon.com
 *
 * Basic idea from
 * Jon Nall - nall@themountaingoats.net - 06.01.04
 *
 * @see file get_dg_matches.pl
 *
 * A. Kempf - a.github@amalesh.de
 * Translation to PHP and all the other stuff, see README.md
 *
 * Released under the GNU GPLv2 License
 *
 */

// loads configuration
$cfg = [];
include "../config.php";
if (file_exists(  "../config.local.php")) {
    include  "../config.local.php";
}

$linkBaseDG = "http://www.dailygammon.com/bg/user/";
$linkExportDG = "http://www.dailygammon.com/bg/export/";
$mainWget = "wget --quiet --header=\"Cookie: USERID={$cfg["user"]}; PASSWORD={$cfg["pass"]}\"";

if ($cfg["user"] === "" || $cfg["pass"] === "") {
    echo "Please set \$cfg['user'] and \$cfg['pass'] in ../config.local.php or ../config.php \n";
    exit;
}


// read thisOutput directory to pretend double analyses
// ************************************************
// get directory handle
$dp = opendir($cfg["pathMat"]);

// read directory and build an array of values
$oldMatches = []; // will contain all existings matches
$newMatches = [];

// our later Output
$thisOutput[] = "";


$i = 0;
while (($fileExist = readdir($dp)) !== false) {
    if ($fileExist !== "." && $fileExist !== ".." && false !== strpos($fileExist, ".mat")) {
        $oldMatches[$i] = str_replace(".mat", "", $fileExist);
        $i++;
    }
}

// get page with finished matches of the last 180 days
$execute = $mainWget . " -O - " . $linkBaseDG . $cfg['user'] . "?finished=1&days_to_view=" . $cfg["daysToView"];

exec($execute, $finished);

// get match numbers
if (!empty($finished) && is_array($finished)) {
    foreach ($finished as $key => $val) {
        if (preg_match("/<A href=(.*?)>Review<\/A>/", $val)) {
            preg_match("/bg\/game\/(\d+)/", $val, $match);
            $newMatches[] = $match[1];
        }
    }
}

// the new matches
$newMatches = array_diff($newMatches, $oldMatches);

$cntNewMatches = count($newMatches);
$cntOldMatches = count($oldMatches);


if ($cntNewMatches || $cntOldMatches) {
    $thisOutput[] = "{$cntOldMatches} existing and {$cntNewMatches} new matches";
}

// get the new match filesShow
foreach ($newMatches as $key => $val) {
    $thisOutput[] = "Getting Game #{$val}";

    $execute = $mainWget . " -O " . $cfg["pathMat"] . $val . ".mat" . " " . $linkExportDG . $val;

    $status = exec($execute);

    if ($status > "") {
        $thisOutput[] = "Error getting Game #{$val} - Status: {$status}";
    }
}

$fullOutput = implode('<br>', $thisOutput);

$tplBody = <<< HTML
<!DOCTYPE html>
<html lang="en" dir="ltr" xmlns="http://www.w3.org/1999/xhtml">
<head>
    <link rel="stylesheet" href="../styles.css" type="text/css"/>
    <title>Downloading *.mat files from dailygammon.com</title>
</head>
<body>
<h1>Downloading *.mat files from dailygammon.com</h1>
<p>{$fullOutput}</p>
<p>Open <a href='write_gnucmd.php'>write_gnucmd</a> to write the gnubg command file</p>
</body>
</html>
HTML;

echo $tplBody;